.github/workflows/upgrade-test.yml

```diff
@@ -0,0 +1,157 @@
+name: DSP Install and Upgrade
+
+on:
+  workflow_dispatch:
+    inputs:
+      released-version:
+        default: 'v1.y.y'
+        description: 'Most recently released DSP version'
+        required: true
+      candidate-version:
+        default: 'v1.x.x'
+        description: 'Candidate version for upgrade'
+        required: true
+
+concurrency:
+  group: ${{ github.head_ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  kubernetes-olm-upgrade:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 60
+    env:
+      OLM_VERSION: v0.24.0
+      VERSION: "v0.0.0-ghaction"  # Need to supply some semver version for bundle to be properly generated
+      CATALOG_BASE_IMG: "registry.access.redhat.com/redhat/community-operator-index:v4.13"
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # fetching also previous commits to get tags
+
+      - name: Set Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: v1.19
+
+      - name: Set up gotestfmt
+        uses: gotesttools/gotestfmt-action@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Not sure if you need this, do you plan to run go lang tests in here?

Reply:
My plan was to run some go tests earlier but writing a basic bash script instead now. Done, took this step out.